via/configuration.py

```diff
@@ -0,0 +1,102 @@
+"""Tools for reading in configuration."""
+
+from inflection import camelize
+
+# pylint: disable=too-few-public-methods
+
+
+class Configuration:
+    """Extracts configuration from params."""
+
+    # Certain configuration options include URLs which we will read, write or
+    # direct the user to. This would allow an attacker to craft a URL which
+    # could do that to a user, so we whitelist harmless parameters instead
+    # From: https://h.readthedocs.io/projects/client/en/latest/publishers/config/#config-settings
+    H_CONFIG_WHITELIST = {
+        # Things we use now
+        "openSidebar",
+        "requestConfigFromFrame",
+        # Things which seem safe
+        "enableExperimentalNewNoteButton",
+        "externalContainerSelector",
+        "focus",
+        "showHighlights",
+        "theme",
+    }
+
+    @classmethod
+    def extract_from_params(cls, params):
+        """Extract Via and H config from query parameters.
+
+        :param params: A mapping of query parameters
+        :return: A tuple of Via, and H config
+        """
+
+        via_params = cls._unflatten(params)
+        h_params = via_params.pop("h", {})
+
+        cls._to_camel_case(h_params)
+        cls._filter_h_params(h_params)
+        cls._move_legacy_params(via_params, h_params)
+
+        # Set some defaults
+        h_params["appType"] = "via"
+        h_params.setdefault("showHighlights", True)
+
+        return via_params, h_params
+
+    @staticmethod
+    def _unflatten(params):
+        """Convert dot delimited flat data into nested dicts."""
+
+        data = {}
+
+        for key, value in params.items():
+            parts = key.split(".")
+            if parts[0] != "via":
+                continue
+
+            target = data
+            # Skip the first 'via' part and then recurse to the one before last
```

Comment:
This is not recursion right? maybe change verb to "iterate"

Reply:
We are iterating over the keys, but recursively descending into the object by repeatedly overwriting `target`. But you're right here, the comment is about the key iteration.

Reply:
I've updated the comment